cms/djangoapps/contentstore/rest_api/v0/views/tabs.py

```diff
@@ -0,0 +1,183 @@
+        return Response(CourseTabSerializer(tabs_to_render, many=True).data)
+
+    @apidocs.schema(
+        body=CourseTabUpdateSerializer(
+            help_text=_(
+                "You can either move tabs or edit a tab to show/hide it."
+                "Only one operation can be performed at a time."
+            )
+        ),
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            204: "In case of success, a 204 is returned with no content.",
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def put(self, request: Request, course_id: str) -> Response:
+        """
+        Show/hide/move tabs in a course.
+
+        **Example Requests**
+
+        Hide a course tab using ``tab_id``:
+
+            PUT /api/contentstore/v0/tabs/{course_id} {
+                "tab_id_locator": {
+                    "tab_id": "wiki"
+                },
+                "is_hidden": true
+            }
+
+        Hide a course tab using ``tab_location``
+
+            PUT /api/contentstore/v0/tabs/{course_id} {
+                "tab_id_locator": {
+                    "tab_location": "block-v1:TestX+DemoX+Demo+type@static_tab+block@796e833d7f6942d3a73a7e50697c1e0d"
+                },
+                "is_hidden": true
+            }
+
+        Move course tabs:
+
+            PUT /api/contentstore/v0/tabs/{course_id} {
+                "tabs": [
+                    {
+                        "tab_id": "info"
+                    },
+                    {
+                        "tab_id": "courseware"
+                    },
+                    {
+                        "tab_locator": "block-v1:TstX+DemoX+Demo+type@static_tab+block@d26fcb0e93824fbfa5c9e5f100e2511a"
+                    },
+                    {
+                        "tab_id": "wiki"
+                    },
+                    {
+                        "tab_id": "discussion"
+                    },
+                    {
+                        "tab_id": "progress"
+                    }
+                ]
+            }
+
+        **Response Values**
+
+        If the request is successful, an HTTP 204 response is returned
+        without any content.
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_write_access(request.user, course_key):
+            raise self.permission_denied(request)
+
+        course_module = modulestore().get_course(course_key)
+        serializer = CourseTabUpdateSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        update_tabs_handler(course_module, serializer.validated_data, request.user)
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
This makes more sense as `POST` and, if possible, returning the new tab state.

Reply:
Since this is not creating anything new, but updating an existing resource `PUT` made most sense to me. Although since it's a partial update perhaps this can be `PATCH` instead.

Reply:
I'm a bit of a REST zealot, so bear with me 😅 : `POST` does not imply creation semantics; in fact, it doesn't imply any semantics, which is why it's used for "anything goes" and generally frowned upon.
I feel it's more appropriate in this case because it feels more like an RPC than an actual REST resource.

I'm not against using `PATCH` here but IMO it's not a good choice because the JSON media type doesn't support the semantics for things like "reorder these elements" or "remove/hide this", so you end up encoding operations into the payload, making it pretty much the same as `POST`.

I don't mean to drag this on but would like to add that, as you mentioned above, the best solution would be to create separate resources that have the correct semantics for what we want to do:
* the "sort" is actually setting the head of the tab list, so you could model it as `PUT /tabs/COURSE_PATTERN/head`
* an update to a single tab would be a `PUT /tabs/COURSE_PATTERN/tab?tab_id=TAB_ID` or PUT `/tabs/COURSE_PATTERN/tab?tab_locator=TAB_LOCATOR`

The TL;DR: here is that I'm ok with both the current solution and either of the suggestions; I think we're in agreement that fleshing out a perfect API model is not the goal at this point 🙂

Reply:
Mainly I want to leave the API open to possibilities like adding / creating a new tab, deleting a tab etc in the future. Which would be better done through those operations.  

It's a pretty small change changing which HTTP verb is used, so I'll accept your suggestion for now, and take in feedback from the edX reviewer as well.